File: MyVinted.Core.Application/Models/LogModel.cs
using System;
using MyVinted.Core.Application.Helpers;

namespace MyVinted.Core.Application.Models
{
    public record LogModel
    (
        DateTime Date,
        string Message,
        string Level = LogLevel.Information,
        string Exception = null,
        string RequestMethod = null,
        string RequestPath = null,
        int? StatusCode = null,
        float? Elapsed = null,
        string SourceContext = null,
        string RequestId = null,
        string ConnectionId = null
    );
}

File: MyVinted.Client/src/app/models/domain/log-model.ts
export interface LogModel {
  date: Date;
  message: string;
  level: string;
  exception: string;
  requestMethod: string;
  requestPath: string;
  statusCode?: number;
  elapsed?: number;
  sourceContext: string;
  requestId: string;
  connectionId: string;
}
